web/source/osk/layouts/targetedFloatLayout.ts

```diff
@@ -1,14 +1,214 @@
+          if(keymanweb.isCJK()) {
+            layout.titleBar.setPinCJKOffset();
+          }
+
+          keymanweb.uiManager.justActivated = true;
+        }
+
+        // Note:  _this.oskView may not be initialized yet.
+        onDragMove(cumulativeX: number, cumulativeY: number) {
+          if(!layout.oskView) {
+            return;
+          }
+
+          layout.titleBar.showPin(true);
+          layout.oskView.userPositioned = true;
+
+          layout.oskView._Box.style.left = (this.startX + cumulativeX) + 'px';
+          layout.oskView._Box.style.top  = (this.startY + cumulativeY) + 'px';
+
+          var r=layout.oskView.getRect();
+          layout.oskView.setSize(r.width, r.height, true);
+          layout.oskView.x = r.left;
+          layout.oskView.y = r.top;
+        }
+
+        onDragRelease() {
+          if(!layout.oskView) {
+            return;
+          }
+
+          let keymanweb = com.keyman.singleton;
+
+          keymanweb.domManager.focusLastActiveElement();
+
+          keymanweb.uiManager.justActivated = false;
+          keymanweb.uiManager.setActivatingUI(false);
+
+          if(layout.oskView.vkbd) {
+            layout.oskView.vkbd.currentKey=null;
+          }
+
+          layout.oskView.userPositioned = true;
+          layout.oskView.doResizeMove();
+          layout.oskView.saveCookie();
+        }
+      }
+
+      return this._moveHandler;
+    }
+
+    private get resizeDragHandler(): MouseDragOperation {
+      const layout = this;
+
+      if(this._resizeHandler) {
+        return this._resizeHandler;
+      }
+
+      this._resizeHandler = new class extends MouseDragOperation {
+        startWidth: number;
+        startHeight: number;
+
+        constructor() {
+          super('se-resize'); // The type of cursor to use while 'active'.
+        }
+
+        onDragStart() {
+          if(!layout.oskView) {
+            return;
+          }
+
+          this.startWidth = layout.oskView.vkbd.kbdDiv.offsetWidth;
+          this.startHeight = layout.oskView.vkbd.kbdDiv.offsetHeight;
+
+          let keymanweb = com.keyman.singleton;
+
+          keymanweb.uiManager.justActivated = true;
+        }
+
+        // Note:  _this.oskView may not be initialized yet.
+        onDragMove(cumulativeX: number, cumulativeY: number) {
+          if(!layout.oskView) {
+            return;
+          }
+
+          let newWidth  = this.startWidth  + cumulativeX;
+          let newHeight = this.startHeight + cumulativeY;
+          
+          // Set the smallest and largest OSK size
+          if(newWidth < 0.2*screen.width) {
+            newWidth = 0.2*screen.width;
+          }
+          if(newHeight < 0.1*screen.height) {
+            newHeight = 0.1*screen.height;
+          }
+          if(newWidth > 0.9*screen.width) {
+            newWidth = 0.9*screen.width;
+          }
+          if(newHeight > 0.5*screen.height) {
+            newWidth = 0.5*screen.height;
```

Comment:
```suggestion
            newHeight = 0.5*screen.height;
```

I am guessing this may fix #5296?

Reply:
I knew it looked wrong, but didn't quite make that connection at the time, and it was the original code.  Good call.

Should be able to 🍒-pick this tidbit without much trouble.

Reply:
🍒: #5428.